File: Java/Exercices/workspaces/solid-isp-start/src/main/java/zenika/solid/isp/MultiFunctionCopier.java
package zenika.solid.isp;

public class MultiFunctionCopier implements Machine {

	@Override
	public void print() {
		System.out.println("Print pages");
	}

	@Override
	public void fax() {
		System.out.println("Fax pages");
	}

	@Override
	public void scan() {
		System.out.println("Scan pages");
	}

	@Override
	public void photocopy() {
		System.out.println("Photocopy pages");
	}

}

File: SOLID/InterfaceSegregation/MultiFunctionCopier.cs
using System;

namespace SOLID.InterfaceSegregation
{
    public class MultiFunctionCopier : IMachine
    {


        public void Print()
        {
            Console.WriteLine("Print pages");
        }

        public void Fax()
        {
            Console.WriteLine("Fax pages");
        }

        public void Scan()
        {
            Console.WriteLine("Scan pages");
        }

        public void Photocopy()
        {
            Console.WriteLine("Photocopy pages");
        }

    }
}
